game-service/src/main/java/org/libertybikes/game/core/MovingObstacle.java

```diff
@@ -89,20 +91,41 @@ public void move(short[][] board) {
         if (yDir != 0) {
             if (yDir > 0) {
                 if (!willCollideY) {
-                    moveDown(board);
+                    moveDown = true;
                 } else if (!hasCollision(board, DIRECTION.UP)) {
-                    moveUp(board);
+                    moveUp = true;
                     yDir = yDir * -1;
                 }
             } else {
                 if (!willCollideY) {
-                    moveUp(board);
+                    moveUp = true;
                 } else if (!hasCollision(board, DIRECTION.DOWN)) {
-                    moveDown(board);
+                    moveDown = true;
                     yDir = yDir * -1;
                 }
             }
         }
+        if (moveRight) {
```

Comment:
instead of creating different methods for each movement, can we instead track the x/y values and then call a new `move(deltaX,deltaY)` method here?